kmip/services/kmip_client.py

```diff
@@ -285,13 +285,16 @@ def open(self):
             six.reraise(*last_error)
 
     def _create_socket(self, sock):
-        self.socket = ssl.wrap_socket(
+        context = ssl.SSLContext(self.ssl_version)
+        context.verify_mode = self.cert_reqs
+        if self.ca_certs:
+            context.load_verify_locations(self.ca_certs)
+        context.check_hostname = False
+        if self.certfile:
```

Comment:
I'm testing combinations of "forgetting" to specify `certfile` or `keyfile` in the configuration file. With this code, if you specify `certfile` but not `keyfile`, you'll get an exception from `load_cert_chain`. If you specify `keyfile` but not `certfile`, this code is skipped and you'll get whatever the SSL stack ends up deciding. The old module-level `wrap_socket` had an additional check

```python
    if keyfile and not certfile:
        raise ValueError("certfile must be specified")
```

to prevent this. Maybe this would be good to have here as well.

Reply:
Done